python/main.py

```diff
@@ -0,0 +1,92 @@
+import functools
```

Comment:
Just wondering now if naming this file `metaparticle` would make more sense.  The import would be something like

`from metaparticle import containerize`

Reply:
Oh! Also good. I'll fix that too.